### ISSUE TITLE
Condition on dependency not respected

### ISSUE THREAD

**Issue by marcadella** [helm/helm#7124]
Setting a condition on a dependency in `Chart.yaml` does not work in Helm v3. [Attached](https://github.com/helm/helm/files/3910568/helmTest2.tar.gz) is an example where a chart has Minio as dependency with the condition `minio.enabled`, which by default is set to `false` in `values.yaml`. When executing `helm template`, Minio resources are present in the output.

Output of `helm version`:
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}

**Comment by karuppiah7890** [comment]
Thanks for trying out helm and reporting this issue. On checking this is what I noticed:

The `values.yaml` content is this
```
minio.enabled: false
```

and `Chart.yaml` mentions condition in dependency as:
```
condition: minio.enabled
```

But I think that's not going to work. Like, yaml keys with dot cannot work (even though it's a valid yaml key), as helm uses the dot notation for the path of the nested keys. So, for above condition, the `values.yaml` needs to be like this :

```
minio:
  enabled: false
```

If you want top level / root level keys (no nested), I think an alternative could be to use camel case, or underscore or may be even hyphenated key name. I tried camel case like the below and it works well

`Chart.yaml`
```
condition: minioEnabled
```

`values.yaml`
```
minioEnabled: false
```

Let me know if that answers the question. I don't think we can consider this as an issue, but yes, it's a drawback that we cannot use the flexibility of yaml (dots in key names) as helm uses it to denote the path in nested keys
